xwiki-commons-core/xwiki-commons-legacy/xwiki-commons-legacy-component/xwiki-commons-legacy-component-default/src/main/java/org/xwiki/component/embed/GenericProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.component.embed;
+
+import javax.inject.Provider;
+
+import org.xwiki.component.internal.AbstractGenericProvider;
+import org.xwiki.component.internal.JavaXGenericProvider;
+import org.xwiki.component.internal.RoleHint;
+import org.xwiki.component.manager.ComponentManager;
+
+/**
+ * Default provider used when the Component Manager needs to inject a {@link Provider} field but no custom Provider has
+ * been registered. The default behavior is simply to look up the Component role class when {@link Provider#get()} is
+ * called. This is useful for example when you wish to do "lazy injection".
+ *
+ * @param <T> the role type
+ * @version $Id$
+ * @since 3.3M2
+ * @deprecated
```

Comment:
Isn't this missing a comment which class should be used instead?

Reply:
This should have never been public in the first place, there are only internal alternatives. I forgot about that one in the pull request description, indeed.

Reply:
I would have just expected at least some comment here for potential (even if unlikely) users of that class what they should do. However, I just checked, and it's probably not used anymore after your change so it seems okay.